### ISSUE TITLE
Incompatible with Spigot 1.16.5

### ISSUE THREAD

**Issue by emgape** [SeanOMik/EnergeticStorage#16]
This plugin is just what I was looking for my server. However, I do not manage to make it work.

When I create an ES system and put it on the floor, I get the following log:

> [13:21:53] [Server thread/ERROR]: Could not pass event BlockPlaceEvent to EnergeticStorage v0.5.1-SNAPSHOT
org.bukkit.event.EventException: null
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:319) ~[spigot.jar:3016-Spigot-73fb609-ea050e6]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[spigot.jar:3016-Spigot-73fb609-ea050e6]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:589) ~[spigot.jar:3016-Spigot-73fb609-ea050e6]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:576) ~[spigot.jar:3016-Spigot-73fb609-ea050e6]
	at org.bukkit.craftbukkit.v1_16_R3.event.CraftEventFactory.callBlockPlaceEvent(CraftEventFactory.java:364) ~[spigot.jar:3016-Spigot-73fb609-ea050e6]
	at net.minecraft.server.v1_16_R3.ItemStack.placeItem(ItemStack.java:220) ~[spigot.jar:3016-Spigot-73fb609-ea050e6]
	at net.minecraft.server.v1_16_R3.PlayerInteractManager.a(PlayerInteractManager.java:497) ~[spigot.jar:3016-Spigot-73fb609-ea050e6]
	at net.minecraft.server.v1_16_R3.PlayerConnection.a(PlayerConnection.java:1356) ~[spigot.jar:3016-Spigot-73fb609-ea050e6]
	at net.minecraft.server.v1_16_R3.PacketPlayInUseItem.a(PacketPlayInUseItem.java:27) ~[spigot.jar:3016-Spigot-73fb609-ea050e6]
	at net.minecraft.server.v1_16_R3.PacketPlayInUseItem.a(PacketPlayInUseItem.java:1) ~[spigot.jar:3016-Spigot-73fb609-ea050e6]
	at net.minecraft.server.v1_16_R3.PlayerConnectionUtils.lambda$0(PlayerConnectionUtils.java:19) ~[spigot.jar:3016-Spigot-73fb609-ea050e6]
	at net.minecraft.server.v1_16_R3.TickTask.run(SourceFile:18) ~[spigot.jar:3016-Spigot-73fb609-ea050e6]
	at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.executeTask(SourceFile:144) ~[spigot.jar:3016-Spigot-73fb609-ea050e6]
	at net.minecraft.server.v1_16_R3.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) ~[spigot.jar:3016-Spigot-73fb609-ea050e6]
	at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.executeNext(SourceFile:118) ~[spigot.jar:3016-Spigot-73fb609-ea050e6]
	at net.minecraft.server.v1_16_R3.MinecraftServer.bb(MinecraftServer.java:942) ~[spigot.jar:3016-Spigot-73fb609-ea050e6]
	at net.minecraft.server.v1_16_R3.MinecraftServer.executeNext(MinecraftServer.java:935) ~[spigot.jar:3016-Spigot-73fb609-ea050e6]
	at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.awaitTasks(SourceFile:127) ~[spigot.jar:3016-Spigot-73fb609-ea050e6]
	at net.minecraft.server.v1_16_R3.MinecraftServer.sleepForTick(MinecraftServer.java:919) ~[spigot.jar:3016-Spigot-73fb609-ea050e6]
	at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:851) ~[spigot.jar:3016-Spigot-73fb609-ea050e6]
	at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$0(MinecraftServer.java:164) ~[spigot.jar:3016-Spigot-73fb609-ea050e6]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_261]
Caused by: java.lang.NullPointerException
	at net.seanomik.energeticstorage.utils.Utils.isBlockASystem(Utils.java:144) ~[?:?]
	at net.seanomik.energeticstorage.listeners.BlockPlaceListener.onBlockPlace(BlockPlaceListener.java:29) ~[?:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_261]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_261]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_261]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_261]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:315) ~[spigot.jar:3016-Spigot-73fb609-ea050e6]
	... 21 more

I also get similar error messages on the server log when I right click the ES system and when I break the block. Clicking the ES system does nothing and when I break it I get a player head instead of the system.

I am using Spigot 1.16.5 and the 0.51 version of the plugin. I have tried to reinstall the plugin, reload it and installing last version of NBT-API, but  nothing fixed the problem. I am a newbie at creating Minecraft servers so maybe I have done anything wrong. 

I uploaded the full server log at pastebin: [https://pastebin.com/krf0UXRx](https://pastebin.com/krf0UXRx)

Thanks for your assistance

**Comment by idnthvn** [comment]
do u have custom textures ?

**Comment by emgape** [comment]
> do u have custom textures ?

No, I do not have any resource packs, nor mods, neither additional plugins.

**Comment by idnthvn** [comment]
Okay. I am running into the same thing. But i have Textures as well. My ES System shows a player head instead of the block. I messed around with the code, not a coder by any means, but i have been dabling in "updating" with errors. I have so far got it to not be "Null" but then i cant get it to read the NBT tags. :P i tried.   but hey, hopefully this guy can update it. I am even willing to try as my server is legit just a server for me and my siblings to play on (4 of us) lol. i also hope that TRANSPORT PIPE plugin also gets updated (someone else forked it) but i think the two plugins can REALLY make it awesome. I played FORGE non-stop, however, my siblings cant afford to have a gaming system like me, so i am building a spigot server just for them to play thats not just vanilla
